osgar/drivers/vanjee.py

```diff
@@ -0,0 +1,63 @@
+"""
+  Handle data from VanJee Lidar WLR-719C (4 lines)
+"""
+import struct
+import math
+
+from osgar.node import Node
+from osgar.bus import BusShutdownException
+
+
+class VanJeeLidar(Node):
+    def __init__(self, config, bus):
+        bus.register('raw', 'xyz', 'scan', 'scan10', 'scanup')
+        super().__init__(config, bus)
+        self.last_frame = None  # not defined
+        self.points = []
+
+    def on_raw(self, data):
+        assert len(data) in [34, 1384], len(data)
+        if len(data) != 1384:
+            return
+        assert data[:2] == b'\xFF\xAA', data[:2].hex()
+        length = struct.unpack_from('>H', data, 2)[0]
+        assert length == 1380, length
+        frame = struct.unpack_from('>H', data, 4)[0]
+        if self.last_frame is not None:
+            assert self.last_frame + 1 == frame, (self.last_frame, frame)
+        self.last_frame = frame
+        assert data[6:8] == bytes.fromhex('0000'), data[6:10].hex()  # reserved
+        # reserved [8:10]  - variable
+        assert data[11:14] == bytes.fromhex('02 190C'), data[11:14].hex()
+        # reserved [14:16]
+        assert data[16:18] == bytes.fromhex('01 04'), data[16:18].hex()
+        bank, motor_speed = struct.unpack_from('>BH', data, 18)
+        assert 1 <= bank <= 16, bank
+        assert 38000 <= motor_speed <= 39300, motor_speed
```

Comment:
Is it known motor speed unit? What about scan resolution? There is fixed points length.

Reply:
I am not sure about the units ... I expect some ticks of encoders. These numbers should correspond to 10Hz. The scan size is on the other hand fixed and resolution 0.1deg? I use 360deg FOV for lidarview and it looks reasonable